ghga_validator/plugins/ref_validation.py

```diff
@@ -0,0 +1,185 @@
+        messages = self.validate_refs(obj, target_class, all_class_ids)
+
+        valid = len(messages) == 0
+
+        result = ValidationResult(
+            plugin_name=self.NAME, valid=valid, validation_messages=messages
+        )
+        return result
+
+    def get_range_class(self, slot_def: SlotDefinition) -> Optional[str]:
+        """Return the range class for a slot
+
+        Args:
+            slot_def (SlotDefinition): Slot Definition
+
+        Returns:
+            Optional[str]: Range class for a slot
+        """
+        all_classes_name = self.schemaview.all_classes().keys()
+        if slot_def:
+            range_class = slot_def.range
+            if range_class in all_classes_name:
+                return range_class
+        return None
+
+    def get_all_class_ids(self, obj: Dict, target_class: str) -> Dict[str, List[str]]:
+        """Get all lists of identifies of inlined objects organized by class name
+
+        Args:
+            obj (Dict): The object to be parsed
+            target_class (str): Target class
+
+        Returns:
+            Dict[class_name, List[str]]: The dictionary containing the lists of
+            identifiers by the class name
+        """
+        all_ids = {}
+
+        for class_name, identifier, _, _ in ObjectIterator(
+            self.schemaview, obj, target_class
+        ):
+            if class_name not in all_ids:
+                all_ids[class_name] = [identifier]
+            else:
+                all_ids[class_name].append(identifier)
+
+        return all_ids
+
+    def validate_refs(
+        self,
+        object_to_validate: Dict,
+        target_class: str,
+        all_class_ids: Dict,
+    ) -> List[ValidationMessage]:
+        """
+        Validate non inlined reference fields in the JSON data
+
+        Args:
+            object_to_validate: input data
+            target_class: parent class in the schema
+            all_class_ids: pre-computed dictionary containing all identifiers ordered by class
+
+        Returns:
+            List[ValidationMessage]: List of validation messages
+
+        """
+        messages = []
+
+        for class_name, _, data, path in ObjectIterator(
+            self.schemaview, object_to_validate, target_class
+        ):
+            for field, value in data.items():
+                slot_def = self.schemaview.induced_slot(field, class_name)
+                range_class = self.get_range_class(slot_def)
+                if range_class and not self.schemaview.is_inlined(slot_def):
+                    non_match = self.find_missing_refs(
+                        value, all_class_ids[range_class]
+                    )
+                    if len(non_match) == 0:
+                        continue
+                    message = ValidationMessage(
+                        severity=SeverityEnum.error,
+                        message="Unknown references "
+                        + f"({self.non_match_as_string(non_match)})",
```

Comment:
```suggestion
                        message="Unknown reference(s) " + str(non_match),
```
Let's keep it simple